### DISCUSSION TITLE
[help] long timestamp formatting is incorrect

### DISCUSSION THREAD

**Discussion by brendanf** [ropensci/targets#1239] [Help]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

The long format timestamps used in the timestamp reporter have incorrect fractional seconds since 61778441.  I noticed this in my pipeline, where the fractional seconds were always exactly the same as the integer seconds, but it's easy to reproduce it without actually running a pipeline.

```R
library(targets)
time <- as.POSIXct("2024-02-20 17:57:13.78")
targets:::time_stamp(time)
#[1] "2024-02-20 15:57:13.13"
targets:::time_stamp_format
#[1] "%Y-%m-%d %H:%M:%S.%OS"
```

The repetition comes because `%OS` includes the seconds as well as the fractional seconds rounded to a certain number of digits past the decimal, by default 0.  I think the format you want is something like `"%Y-%m-%d %H:%M:%OS2"`.

**Comment by wlandau** [comment]
Fixed as of 13f74aaf5c4c779489ed9040c5372b99c6496a76
